packages/plugin-essentials/sources/commands/install.ts

```diff
@@ -291,7 +295,7 @@ export default class YarnCommand extends BaseCommand {
     }
 
     const {project, workspace} = await Project.find(configuration, this.context.cwd);
-    const cache = await Cache.find(configuration, {immutable: this.immutableCache, check: this.checkCache});
+    const cache = await Cache.find(configuration, {immutable: this.immutableCache && !updateMode, check: this.checkCache});
```

Comment:
Small nit: I think it would be better to extract the `immutable` property of the `Cache.find` parameter next to the `immutable` declaration:
```ts
const immutable = ...;
const immutableCache = this.immutableCache && !updateMode;
```